### DISCUSSION TITLE
[Feature] Slideshow Viewing

### DISCUSSION THREAD

**Discussion by dercavendar1** [immich-app/immich#4084] [Feature Request]
### The feature

It would be a great feature if there could be some sort of "digital picture frame" feature in the app and/or web.  Something like a randomized slideshow (even better if you could enter parameters to limit which photos)

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by alextran1502** [comment]
Hello, you can already do this on the web. I am planning to add this on the mobile at some point as well. 

<img width="1686" alt="image" src="https://github.com/immich-app/immich/assets/27055614/068c14eb-38c5-44f8-a33c-75786472ca3c">

**Reply by radh21301** [reply]
Even I am doing this. I have a certain album that I have for my digital photo frame. But would really like the option to randomize the pictures and have some control over the rate at which those pictures change. The problem with no randomization is that if the slide show gets interrupted, it starts from the 1st picture again. Also, a random order keeps things interesting! I think this makes it a perfect feature for a digital photo frame, but not super important for people normally looking at their pictures I guess.

**Reply by RyGuy994** [reply]
With this implementation in the future for the Mobile app. I am currently using foto because I can't get it in immich yet. Can there be an option to Start in slideshow mode and also in a shared album.

Use Case: My wife, kids, and I have a shared album that we want to display on a few digital photo frames. We would like the option when we start the app to automatically go to the shared album and start slideshow. Randomizer would be nice feature as well!!!

~lover of immich

**Comment by dercavendar1** [comment]
Thank you for the quick response.

I did see that, but there is no option to randomize the slideshow or limit
based on anything (maybe I just want my cats today etc.)

**Reply by jrasm91** [reply]
You can make an album and do the slideshow from there if you want only select photos in it.

**Comment by nickstrand** [comment]
Is there a way to trigger the slideshow directly from a shared url, like `?slideshow` being tacked onto the url?  There's some software that is ok with just getting a url.  I agree with this though, it would be nice to be able to use an album on a photo frame

**Reply by miloszlipski** [reply]
I'm looking for exactly for the same thing - I want start slideshow for photo frame directly from url

**Reply by nickstrand** [reply]
I've got it working with a python script behind an nginx proxy that is redirecting `immichurl/random` to a docker container that does a 302 redirect to a random image url from a shared album.  The code is really sloppy and would require some amount of know-how to get up and running, https://github.com/nickstrand/Immich-python-slideshow.  I'll polish it up someday.  or not.  either way it's doing what I wanted, showing random images from a shared album until they're all gone, refilling, and doing it again.

**Reply by JW-CH** [reply]
Hey @nickstrand,
thank you for your repo. What photoframe are you using? Or are you just using the mobile app?

**Reply by nickstrand** [reply]
I'm using home assistant targeting a url with wallpanel, but anything web browser related would work I would think

**Comment by RyGuy994** [comment]
Well, in a attempt to try to solve this issue. I myself have started playing with docker and will be taking a fork of this to run and see what I can play around with.

**Comment by PersistentCloud** [comment]
I created a react app for this: https://github.com/PersistentCloud/immich-slideshow

Contributions are very appreciated. For example I cannot get a proper cross-fade running.

**Comment by hthornwell** [comment]
Is adding a slideshow feature to the mobile apps still somewhere on the development roadmap?

**Reply by bo0tzz** [reply]
It's not on the roadmap, but that doesn't mean it won't happen.

**Comment by enlight3d** [comment]
Hello, I've seen the slideshow option on photos, but is there a way to have it on the albums (and shared albums too ?)

**Comment by ElderBlade** [comment]
Hi! Any chance the slideshow feature from the web app could be added to mobile? It would be really useful and help bring the mobile app closer to feature parity with web. Thanks for all your work on Immich!

**Comment by pktrance** [comment]
+1 for the slideshow feature to be in the mobile app. 

This is the one thing I always wish Google Photos had. It's great that the feature is already on the Immich web app, but having it on the mobile app would be huge

**Comment by nefariouslegion** [comment]
Just writing to also request a slideshow feature in the mobile apps please!
